static/js/panels.js

```diff
@@ -50,7 +50,9 @@ export function should_show_notifications(ls) {
         // if permission has not been granted yet.
         !notifications.granted_desktop_notifications_permission() &&
         // if permission is allowed to be requested (e.g. not in "denied" state).
-        notifications.permission_state() !== "denied"
+        notifications.permission_state() !== "denied" &&
+        // it doesn't make sense to ask web_public_visitors for notifications access
+        !page_params.is_web_public_visitor
```

Comment:
Can you add node tests verifying this bit of logic?  `frontend_tests/node_tests/panels.js` is reasonably complete.

I'd also put this check first, rather than last.